File: Toggly.FeatureManagement.ReactNative/libs/core/src/services/MemoryStorage.ts
import type { TogglyStorage } from '../models';

/**
 * In-memory storage implementation.
 * Used as a fallback when no storage provider is configured.
 * Data is lost when the app is closed.
 */
export class MemoryStorage implements TogglyStorage {
  private storage: Map<string, string> = new Map();

  async get(key: string): Promise<string | null> {
    return this.storage.get(key) ?? null;
  }

  async set(key: string, value: string): Promise<void> {
    this.storage.set(key, value);
  }

  async delete(key: string): Promise<void> {
    this.storage.delete(key);
  }

  /**
   * Clear all stored data
   */
  clear(): void {
    this.storage.clear();
  }

  /**
   * Get all keys in storage
   */
  keys(): string[] {
    return Array.from(this.storage.keys());
  }
}

File: Toggly.FeatureManagement.iOS/TogglyCore/Sources/Storage/MemoryStorage.swift
import Foundation

/// In-memory storage implementation for Toggly.
/// Data is lost when the app terminates.
public actor MemoryStorage: TogglyStorage {
    private var storage: [String: String] = [:]

    /// Creates a new in-memory storage instance.
    public init() {}

    /// Retrieve a value from storage.
    public func get(_ key: String) async -> String? {
        return storage[key]
    }

    /// Store a value.
    public func set(_ key: String, value: String) async {
        storage[key] = value
    }

    /// Delete a value from storage.
    public func delete(_ key: String) async {
        storage.removeValue(forKey: key)
    }

    /// Clear all stored values.
    public func clear() async {
        storage.removeAll()
    }
}
